doc/ui-guidelines.md

```diff
@@ -0,0 +1,252 @@
+
+BEFORE:
+```clojure
+(defn- view-internal
+  [_ _]
+  (let [pressed? (reagent/atom false)]
+    (fn
+      [{:keys [theme on-press on-long-press icon]}]
+      [rn/pressable
+       {:style               (style/main @pressed? theme)
+        :on-press            on-press
+        :on-press-in         #(reset! pressed? true)
+        :on-press-out        #(reset! pressed? nil)
+        :on-long-press       on-long-press}
+       [quo.icons/icon icon]])))
+
+(def view (theme/with-theme view-internal))
+```
+
+NOW:
+```clojure
+(defn view
+  [{:keys [on-press on-long-press icon]}]
+  (let [[pressed? set-pressed] (rn/use-state false)
+        theme                  (theme/use-theme-value)
+        on-press-in            (rn/use-callback #(set-pressed true))
+        on-press-out           (rn/use-callback #(set-pressed nil))]
+    [rn/pressable
+     {:style               (style/main pressed? theme)
+      :on-press            on-press
+      :on-press-in         on-press-in
+      :on-press-out        on-press-out
+      :on-long-press       on-long-press}
+     [quo.icons/icon icon]]))
+```
+
+
+- We no longer need to create an anonymous function for rendering. This removes unnecessary confusion and the need for specific knowledge on how it works and why it was needed.
+- `rn/use-state` is used instead of `reagent/atom`
+- State values no longer need to be dereferenced; they are accessible as regular symbols. This eliminates a common bug where the "@" symbol was inadvertently omitted.
+- `theme/with-theme` wrapper is not needed anymore, `(theme/use-theme-value)` hook can be used directly in the components
+- `:f>` not needed anymore, all components are functional by default
+- `rn/use-callback` hook should be used for anon callback functions
+
+> [!IMPORTANT]
+> DO NOT USE anon functions directly in the props
+ 
+BAD
+```clojure
+(defn view
+  []
+  (let [[pressed? set-pressed] (rn/use-state false)]
+    [rn/pressable
+     {:style               (style/main pressed?)
+      :on-press-in         #(set-pressed true)
+      :on-press-out        #(set-pressed nil)}]))
+```
+
+GOOD:
+```clojure
+(defn view
+  []
+  (let [[pressed? set-pressed] (rn/use-state false)
+        on-press-in            (rn/use-callback #(set-pressed true))
+        on-press-out           (rn/use-callback #(set-pressed nil))]
+    [rn/pressable
+     {:style               (style/main pressed?)
+      :on-press-in         on-press-in
+      :on-press-out        on-press-out}]))
+```
+
+## Global state and subscriptions
+
+For global state management, we utilize Re-frame subscriptions. They can be likened to React state. To obtain the state, `(rf/sub [])` is employed, and to modify it, `(rf/dispatch [])` is utilized. However, they update components in a similar manner to React states.
```

Comment:
`They can be likened to React state`
typo, linked?

Reply:
no, it's like compared, similar